microcosm_sqlite/factories.py

```diff
@@ -3,37 +3,54 @@
 
 """
 from distutils.util import strtobool
+from pkg_resources import iter_entry_points
 
 from microcosm.api import defaults
 from sqlalchemy import create_engine
 from sqlalchemy.orm import sessionmaker
 
 
 @defaults(
-    path=":memory:",
-    # echo all SQL
     echo="False",
+    path=":memory:",
+    paths=dict(),
 )
-def configure_sqlite_engine(graph):
+class SQLiteBindFactory:
+    """
+    A factory for SQLite engines and sessionmakers based on a name.
+
     """
-    Configure the SQLite engine for SQLAlchemy.
+    def __init__(self, graph):
+        self.default_echo = strtobool(graph.config.sqlite.echo)
+        self.default_path = graph.config.sqlite.path
 
-    The primary input is the connection URL path, which is assumed to have the form:
+        self.engines = dict()
+        self.sessionmakers = dict()
+        self.paths = {
+            entry_point.name: entry_point.load()()
+            for entry_point in iter_entry_points("microcosm.sqlite")
```

Comment:
Entry point support means that a library can define a path for its database file.